core/kazoo_ast/src/fs_prop_usage.erl

```diff
@@ -0,0 +1,648 @@
+    sets:from_list(
+      lists:usort(
+        lists:filter(fun is_binary/1
+                    ,lists:foldl(fun usage_keys/2, [], Usages)
+                    )
+       )
+     ).
+
+usage_keys({'props', 'get_value', Key, _VarName, _Default}, Acc) ->
+    [Key |Acc];
+usage_keys({'props', 'get_integer_value', Key, _VarName, _Default}, Acc) ->
+    [Key |Acc];
+usage_keys({'props', 'get_binary_value', Key, _VarName, _Default}, Acc) ->
+    [Key |Acc];
+usage_keys({'props', 'get_is_true', Key, _VarName, _Default}, Acc) ->
+    [Key |Acc];
+usage_keys({'props', 'is_true', Key, _VarName, _Default}, Acc) ->
+    [Key |Acc];
+usage_keys({'props', 'get_first_defined', Keys, _VarName, _Default}, Acc) ->
+    Keys ++ Acc;
+usage_keys({'props', 'delete_keys', Keys, _VarName, _Default}, Acc) ->
+    Keys ++ Acc;
+usage_keys({'props', 'filter', 'undefined', _VarName, _Default}, Acc) -> Acc;
+usage_keys({'props', 'insert_values', _Keys, _VarName, _Default}, Acc) -> Acc;
+usage_keys({'props', 'set_values', _Values, _VarName, 'undefined'}, Acc) -> Acc.
+
+event_filter_filename() ->
+    filename:join([code:lib_dir('ecallmgr', 'src')
+                  ,"fs_event_filters.hrl"
+                  ]).
+
+-record(usage, {usages = [] %% places the Data is accessed
+               ,data_var_name = 'Props' %% Tracks current var name
+               ,data_var_aliases = [] %% typically when kz_json:set_value is used
+               ,current_module %% what module are we currently in
+               ,functions = [] %% AST functions loaded
+               ,visited = [] %% MFAs visited (to stop recursion)
+               }).
+
+-spec process() -> [{module(), list()}].
+process() ->
+    io:format("processing ecallmgr FreeSWITCH Props usage: "),
+    Usages = [{Module, Usages} ||
+                 Module <- kz_ast_util:app_modules('ecallmgr'),
+                 (Usages = process(Module)) =/= 'undefined'
+             ],
+    io:format(" done~n"),
+    Usages.
+
+-spec process(module()) -> list().
+process(Module) when is_atom(Module) ->
+    U = process_action(Module),
+    io:format("."),
+    ?DEBUG("  usage for ~p: ~p~n", [Module, U]),
+    U.
+
+process_action(Module) ->
+    FArgs = function_args(Module),
+    process_action(Module, FArgs).
+
+process_action(Module, {_F, _As}=FArg) ->
+    process_action(Module, [FArg]);
+process_action(Module, FArgs) ->
+    case lists:foldl(fun({F, Args}, Acc) ->
+                             process_action(Module, F, Args, Acc)
+                     end
+                    ,[]
+                    ,FArgs
+                    )
+    of
+        [] -> 'undefined';
+        Usages -> Usages
+    end.
+
+
+process_action(_M, 'undefined', _As, Acc) -> Acc;
+process_action(Module, Function, Args, Acc) ->
+    #usage{usages=Us} = process_mfa_call(#usage{current_module=Module}
+                                        ,Module, Function, Args
+                                        ),
+    Us ++ Acc.
+
+%% define entry points for modules
```

Comment:
At the moment, these are the function(s) in the modules that provided the best way to find keys used to inspect the proplist. Most, if not all, are functions that will likely not change at this point. I would expect the code to crash if the function signature changes.

Another thought I had was improving our naming in ecallmgr modules to denote FreeSWITCH proplists as `FSProps` versus `gen_listener`-provided proplists (`GLProps`?). Then the code could look for any function that has an arg of `FSProps` and go from there, thus removing the need to have module-specific filters.